torch/csrc/jit/codegen/cuda/test/test_gpu_external_src.cpp

```diff
@@ -0,0 +1,90 @@
+#if defined(USE_CUDA)
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <torch/csrc/jit/codegen/cuda/executor_utils.h>
+#include <torch/csrc/jit/codegen/cuda/fusion.h>
+#include <torch/csrc/jit/codegen/cuda/test/test_gpu_validator.h>
+#include <torch/csrc/jit/codegen/cuda/test/test_utils.h>
+
+// Tests go in torch::jit
+namespace torch {
+namespace jit {
+
+using namespace torch::jit::fuser::cuda;
+
+// This is for internal testing only and is intended to be used as a template to
+// compile and run an external source file. By default, it should just
+// return immediately, but if PYTORCH_NVFUSER_EXTERNAL_SRC is defined,
+// the file specified by the env var is loaded and compiled.
+TEST_F(NVFuserTest, FusionExternalSrc_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+  FusionExecutor fe;
+
+  // By default, this env var should not be defined. To test using an
+  // external source file, set it to the path to the external source
+  // file.
+  auto path = std::getenv("PYTORCH_NVFUSER_EXTERNAL_SRC");
```

Comment:
nit: Should we prefer `NVFUSER_EXTERNAL_SRC` over `PYTORCH_NVFUSER_EXTERNAL_SRC`? We are pulling submodule and becoming a standalone project.

Reply:
I think we should still keep using the `PYTORCH_` prefix as we are still inside the PyTorch codebase. We could easily do find-and-replace once moved out.